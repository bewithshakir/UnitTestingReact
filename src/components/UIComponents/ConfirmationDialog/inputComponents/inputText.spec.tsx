import { shallow } from 'enzyme';
import InputText from './inputText';


const mockFormik = {
    touched: {},
    error: "",
    errors: {},
    initialError: "",
    initialTouched: false,
    initialValue: "",
    values: {
        text: '',
    },
    getFieldProps: jest.fn(),
    setFieldValue: jest.fn()
};

describe('renders inputText properly', () => {
    const onChange = jest.fn();
    const handleTouch = jest.fn();
    const inputComponent = shallow(
        <InputText field={{
            fieldType: 'text',
            name: 'text',
            label: 'Enter Name',
        }}
            fieldId={`textInput`}
            formik={mockFormik as any}
            onChange={onChange}
            handleTouched={handleTouch}
        />);

    it('snapshot testing', () => {
        expect(inputComponent).toMatchSnapshot();
    });
});
