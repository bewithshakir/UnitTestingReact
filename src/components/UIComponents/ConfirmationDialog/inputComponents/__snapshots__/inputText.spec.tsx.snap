// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inputText properly snapshot testing 1`] = `
<Input
  autoComplete="new-password"
  disabled={false}
  error={false}
  id="textInput"
  key="textInput"
  label="Enter Name"
  name="text"
  onBlur={[Function]}
  onChange={[Function]}
  required={false}
  type="text"
  value=""
  width="100%"
/>
`;
